### DISCUSSION TITLE
Questions about Compose and Data Augmentation

### DISCUSSION THREAD

**Discussion by pauljhin** [Project-MONAI/tutorials#1032] [Q&A] [answered]
Hello!

In the [spleen_segmentation_3d_visualization_basic.ipynb](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d_visualization_basic.ipynb) tutorial in one part it says: "Here we use several transforms to augment the dataset". The ``` Compose ``` is used with some transformations.

Is ``` Compose ``` for chaining transformations? Right?

My question is:  Does it really increase the data? 

It seems to me that it just applies the transformations but doesn't keep a copy of the data or copy with different transformations. I am wrong?

For example, during training I would like ``` Compose ``` output to look like this:

original_image
original_image_with_rotation
original_image_with_flip_and_noise

How can I do this? Is it possible to have different ``` CacheDatasets ``` with different transformations and merge into just one ``` DataLoader ```?

**Comment by KumoLiu** [comment, marked as answer]
Hi @pauljhin, random transforms will generate different images randomly for each epoch so it can augment the dataset. If you want images with random rotation or random flip, you can use transforms such as `RandFlip`, `RandRotate`, `RandGaussianSmooth`.
Hope it can help you, thanks!

**Reply by pauljhin** [reply]
Hi @KumoLiu,

Now I understand, it wasn't clear that these transformations were made in each epoch.

One last doubt. ```Compose``` does the operations in the order they are in the list? Ex. Compose([FirstTransf -> SencondTransform]). Like a pipeline, the previous output is the input of the next transformation. Right?

**Reply by KumoLiu** [reply]
Hi @pauljhin, yes you are right!
https://github.com/Project-MONAI/MONAI/blob/663cfd0dc806a21ecc3517f4c61823f2581466ab/monai/transforms/compose.py#L39-L41

**Reply by pauljhin** [reply]
Ok Thank you very much @KumoLiu !
